solidity/contracts/cryptography/AltBn128.sol

```diff
@@ -335,6 +363,12 @@ library AltBn128 {
         );
     }
 
+    function _gfP2Add(uint256 ax, uint256 ay, uint256 bx, uint256 by)
```

Comment:
Why do we add `_` suffix for some functions? Solidity allows for function name overloading. I did a test and renamed all `_gfP2*` to `gfP2*` and gas cost of all tests in `TestAltBn128` is the same. Also, the execution time of tests is not any different from what I see.

Reply:
The prefix is to denote stack-only versions of the functions. They're semi not intended for external use because of the more annoying API, but they save gas when used inside loops in the library.